pkg/ring/ring.go

```diff
@@ -100,30 +101,34 @@ func New(cfg Config) (*Ring, error) {
 		return nil, err
 	}
 
+	if cfg.Name == "" {
+		cfg.Name = "ingester"
+	}
```

Comment:
Lets not do this.  Also, I don't think the name is part of the config - I think they're defined by the code.  Name should be a string passed to the constructor, and should not be the empty string.